### PR TITLE
Refactor form modal | Closes #2383

### LINKED ISSUE #2383: Refactor FormModal + Form Structure
In #2374 I converted the `FormModal` component to use antd instead of blueprint, but that causes issues with how our component tree is structured.

Currently our tree looks like this:

```
├── Modal
|	├── Form (RFF wrapper)
|	|	├── FormModalForm (handles form rendering)
|	|	|	├── actual form
```

The issue is that the antd Modal component wants to add a row of buttons along the bottom of the Modal that can be used for closing the modal and/or executing some sort of action (such as submitting a form), but the form submission and form reset actions are not available until you get to the FormModalForm component. 

So we likely need to restructure things to be like:

```
├── Form (RFF wrapper)
|	├── FormModalForm (handles form rendering)
|	|	├── Modal 
|	|	|	├── actual form
```

which can maybe also allow us to simply control the usage of the Modal as part of the form config (ie: via a `openInModal` prop) but that's just one idea.

Anyways, I'll leave this up to you to fix as you see best.  If you have other ideas for cool things we can do, then plz discuss in chat first.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Refactors FormModal to make it aware of RFF
- Updates form fields to set `initialValue` to be used for reset.
- Closes #2383

### CODE REVIEW

File: assets/src/application/ui/forms/formModal/FormModal.tsx [hunk 1]
```diff
@@ -15,39 +14,22 @@ const FormModal: React.FC<FormModalProps> = ({
...
-						<FormModalForm {...formProps} FormComponent={FormComponent} onClose={onCloseHandler} />
-					)}
-				/>
-			</Modal>
+			<Form
+				initialValues={initialValues}
+				onSubmit={onSubmit}
+				{...extraProps}
+				render={({ ...formProps }) => (
+					<FormModalForm {...formProps} FormComponent={FormComponent} onClose={onCloseHandler} />
```
Comment (nicolad) [inline assets/src/application/ui/forms/formModal/FormModal.tsx]: It's nice that RFF exposes `render` prop.

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx [hunk 2]
```diff
@@ -23,6 +18,20 @@ const dataStyle: CSSProperties = {
...
+	minHeight: '50%',
+	minWidth: '320px',
+	width: '80%',
+	overflowY: 'scroll',
+	position: 'absolute' as 'absolute',
+	left: '50%',
+	top: '50%',
+	transform: 'translate(-50%, -50%)',
+};
+
```
Comment (nicolad) [inline assets/src/application/ui/forms/formModal/FormModalForm.tsx]: We need to discuss about this approach at some point.
Not related to this PR, though.

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx [hunk 3]
```diff
@@ -45,42 +54,52 @@ const FormModalForm: React.FC<FormModalFormProps> = ({
...
+					<FormComponent
+						form={form}
+						values={values}
+						submitting={submitting}
+						pristine={pristine}
+						formReset={formReset}
+						{...formProps}
+					/>
+				</div>
+				<pre style={dataStyle}>{JSON.stringify(values, null, 2)}</pre>
```
Comment (nicolad) [inline assets/src/application/ui/forms/formModal/FormModalForm.tsx]: is this a temp ?

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx [hunk 4]
```diff
@@ -45,42 +54,52 @@ const FormModalForm: React.FC<FormModalFormProps> = ({
...
-			<pre style={dataStyle}>{JSON.stringify(values, null, 2)}</pre>
-		</form>
+		<Modal
+			visible={true}
+			onOk={onClose}
+			style={modalStyle}
+			okText={__('Submit')}
+			okButtonProps={submitButton}
+			cancelText={__('Reset')}
+			cancelButtonProps={resetButton}
```
Comment (nicolad) [inline assets/src/application/ui/forms/formModal/FormModalForm.tsx]: cool prop

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx (hunk 3 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/forms/formModal/FormModalForm.tsx]: Yes

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx [hunk 5]
```diff
@@ -45,42 +54,52 @@ const FormModalForm: React.FC<FormModalFormProps> = ({
...
+			onCancel={onClose}
+			width={'80%'}
```
Comment (nicolad) [inline assets/src/application/ui/forms/formModal/FormModalForm.tsx]: perhaps we can set this value as default and allow to configure it, if needed

Comment (manzoorwanijk) [reply assets/src/application/ui/forms/formModal/FormModalForm.tsx]: If you don't pass it, AntD will force set it to `520px`

File: assets/src/application/ui/forms/formModal/FormModalForm.tsx (hunk 2 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/forms/formModal/FormModalForm.tsx]: Agreed

**Pull request merged**